server/NewBankClientHandler.java

```diff
@@ -23,46 +29,64 @@ public void run() {
-						Customer customer = bank.checkLogInDetails(userName, password, this);
-						// If the user is authenticated then get requests from the user and process them.
-						if (customer != null) {
-							out.println("\nLoading...\n");
-							out.println("Log In Successful.\n");
-							out.println("Welcome to your NewBank account " + customer.getUsername()+".");
-							out.println("\n+---------------------------------------------------+");
-							out.println("|NewBank - Customer Menu                            |");
-							out.println("+---------------------------------------------------+");
-							out.println("|(1) - Show existing accounts and current balance   |");
-							out.println("|(2) - Create a new account with zero opening funds |");
-							out.println("|(3) - Move funds between your existing accounts    |");
-							out.println("|(4) - Pay an existing NewBank customer             |");
-							out.println("+---------------------------------------------------+");
-							while (true) {
-								String request = in.readLine();
-								System.out.println("Request from " + customer.getUsername());
-								// Added pass newBankClientHandler Object into processRequest, so it can call methods
-								String response = bank.processRequest(customer, request, this);
-								out.println(response);
+
+						//If 3 failed login attempts during session, does not allow login
+						if (failedLogInCount >2) {
+							out.println("There have been too many unsuccessful login attempts, " +
+									"please disconnect and try again later.");
+							break;
+						}
+
+						while (failedLogInCount<3) {
+							// Ask for username
+							out.println("Enter Username:");
+							String userName = in.readLine();
+
+							/*
+							checks if account is locked, note will only be useful once database is running and
+							the database stores accountLocked variable. Currently resetting connection resets
+							the customer HashMap
+							*/
+							if (isAccountLocked(userName)){
+								out.println("This account is locked.");
+								break;
+							}
+							// Ask for password
+							out.println("Enter Password:");
+							String password = in.readLine();
+							out.println("\nChecking Details...");
+
+							// Authenticate user and get customer ID token from bank for use in subsequent requests.
+							Customer customer = bank.checkLogInDetails(userName, password, this);
+
+							// If the user is authenticated then get requests from the user and process them.
+							if (customer != null) {
+								showCustomerMenu(customer);
+								while (true) {
+									String request = in.readLine();
+									System.out.println("Request from " + customer.getUsername());
+									// Added pass newBankClientHandler Object into processRequest, so it can call methods
+									String response = bank.processRequest(customer, request, this);
+									out.println(response);
```

Comment:
Here you use just the `out.println`, and it goes like this through the function. You should decide which version you want to use, and then be consistent with it, as it can cause confusion